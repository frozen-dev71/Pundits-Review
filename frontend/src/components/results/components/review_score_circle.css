/*
// ======================================
// CSS to style review_score_circle
// ======================================
*/


.review-score-circle {
    border-radius: 2em;
    width: 2em;
    height: 2em;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.35em;
    justify-self: center;
    grid-row: 2/4;
    grid-column: 3/4;
    align-self: center;
}

.green-circle {
	background-color: #006f3c;
}

.yellow-circle {
	background-color: #f9a73e;
}

.red-circle {
	background-color: #bf212f;
}

.black-circle {
    background-color: black;
    opacity: .5;
}




/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 1000px) {


    .review-score-circle {
        font-size: 1.75em;
    }

}