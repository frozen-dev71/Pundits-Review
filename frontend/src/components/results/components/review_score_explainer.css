/*
// ======================================
// CSS for Review Score Explainer
// ======================================
*/

.review-score-explainer-wrapper {
	display: flex;
	justify-content: center;
	align-items: center;
	padding: 1em 0;
	transform: scale(0.6);
}

.review-score-explainer-text {
	font-size: .95em;
	letter-spacing: .2em;
	margin-left: 1em;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 1000px) {

	.review-score-explainer-text {
		font-size: 1.25em;
	}
}
