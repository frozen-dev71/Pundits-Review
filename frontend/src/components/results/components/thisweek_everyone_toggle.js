// ======================================
// This file contains the PLAYER/CLUB toggle menu that is rendered on the RESULTS PAGE
// Allows users to toggle between seeing results for players with review scores THIS WEEK and EVERYONE 
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports
import { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus } from "../../actions/actions";

// CSS Imports 
import './thisweek_everyone_toggle.css';


// ===========
// ThisWeekEveryoneToggle Component 
// ===========

export class ThisWeekEveryoneToggleContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			thisWeekClassName: this.setThisWeekEveryoneClass("THIS WEEK"),
			everyoneClassName: this.setThisWeekEveryoneClass("EVERYONE")
		}
	}


	setThisWeekEveryoneClass = (category) => {

		if (this.props.thisWeekEveryoneToggle === category) {

			return "this-week-everyone-button this-week-everyone-button-active body-text-light"

		} else {

			return "this-week-everyone-button body-text-light"
		}

	}

	_handleClick = (event) => {

		let searchTerm = this.props.searchTerms[this.props.searchTerms.length - 1];

		// Clear Previous Search Results
		this.props.clearSearchResults();

		if (event.target.id === "this-week-button") {

			// Search By Player
			this.props.searchByPlayer(searchTerm);

			// Search By Club
			this.props.searchByClub(searchTerm);

			// Set Redux store toggle status
			this.props.setThisWeekEveryoneToggleStatus("THIS WEEK")

		} 

		if (event.target.id === "everyone-button") {

			// Search By Player
			this.props.searchByPlayer(searchTerm, true);

			// Search By Club
			this.props.searchByClub(searchTerm, true);

			// Set Redux store toggle status
			this.props.setThisWeekEveryoneToggleStatus("EVERYONE")

		}

		// Set the new className for each button 
		this.setThisWeekEveryoneClass();
		
	}

	render() {

		let result = (


			<div className="this-week-everyone-toggle-wrapper">

				<div className="this-week-everyone-button-container">

					<button id="this-week-button" className={this.state.thisWeekClassName} onClick={(e) => this._handleClick(e)}>THIS WEEK</button>
					<button id="everyone-button" className={this.state.everyoneClassName} onClick={(e) => this._handleClick(e)}>ALL WEEKS</button>

				</div>

			</div>

			);

		return result
	}
}



// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return {
  		searchTerms: state.searchTerms,
  		searchDataClubs: state.searchDataClubs,
  		searchDataPlayers: state.searchDataPlayers,
  		thisWeekEveryoneToggle: state.thisWeekEveryoneToggle
  	};
};


export const ThisWeekEveryoneToggle = connect(mapStateToProps, { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus })(ThisWeekEveryoneToggleContainer);

