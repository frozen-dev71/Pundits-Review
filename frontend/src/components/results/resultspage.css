/*
// ======================================
// Core CSS for results page
// ======================================
*/


.results-page-wrapper {
	display: flex;
	flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    height: 100vh;
}


.search-bar-results-wrapper {
	position: absolute;
    display: flex;
    flex-direction: column;
	width: 95%;
	top: 12.5%;
    justify-content: flex-start;
    align-items: center;
    background-color: white;
}


.results-info-text {
    display: none;
	margin: 2vh 8vw;
    align-self: flex-start;
    position: absolute;
    top: 5vh;
    z-index: -1;
}