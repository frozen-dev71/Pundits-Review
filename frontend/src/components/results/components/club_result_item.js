// ======================================
// Club Result Item Component - Contains info on each club returned from the API & links to individual club pages
// ======================================


// ===========
// IMPORTS 
// ===========

import React, { Component } from "react";
import { Link } from 'react-router-dom';
import { connect } from "react-redux";


// Local JS Imports
import { ReviewScoreCircle } from './review_score_circle';
import { badgeFinder } from '../../global/functions/set_data_helpers';
import { stadiumNames } from '../../global/resources/club_stadiums';
import { setClub } from '../../actions/actions';

// CSS Imports
import './result_item.css';

// ===========
// Player Result Container
// ===========


export class ClubResult extends Component {

	_handleClick(clubName) {

		this.props.setClub(this.props.clubName);
	}

	render() {

		let clubNameParam = "/clubs/" + encodeURIComponent(this.props.clubName);

		let result = (

				<Link to={clubNameParam} className="club-results-item-wrapper react-link" onClick={() => this._handleClick(this.props.clubName)}>

					<p className="club-result-club-name heading-text"> {this.props.clubName} </p>
					<span className={"club-badge-clubs-result-page " + badgeFinder(this.props.clubName)}></span>
					<p className="club-result-club-stadium body-text-light"> {stadiumNames[this.props.clubName]} </p>
					<ReviewScoreCircle reviewScore={this.props.reviewScore}/>
					<p className="club-result-view-club body-text-light black-text">VIEW CLUB</p>
					<i className="arrow-club-result-item fa fa-long-arrow-right" aria-hidden="true"></i>

				</Link>
			);

		return result;
	}
}



// ===============
// Connect redux state container to Club Result component
// ===============


const mapStateToProps = state => {
  return { 

	  	clubsViewed: state.clubsViewed
  	};
};



// Export connected component
export const ClubResultContainer = connect(mapStateToProps, { setClub })(ClubResult);