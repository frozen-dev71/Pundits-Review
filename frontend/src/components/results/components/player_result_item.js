// ======================================
// Player Result Item Component - Contains info on each player returned from the API & links to individual player pages
// ======================================


// ===========
// IMPORTS 
// ===========

import React, { Component } from "react";
import { Link } from 'react-router-dom';
import { connect } from "react-redux";

// Local JS Imports 
import { ReviewScoreCircle } from './review_score_circle';
import { setPlayer } from "../../actions/actions";
import { club_shirts } from '../../global/resources/club_shirts';

// CSS Imports
import './result_item.css';

// ===========
// Player Result Container
// ===========


export class PlayerResult extends Component {

	_handleClick(playerName) {

		this.props.setPlayer(this.props.playerName);

	}

	render() {

		let playerNameParam = "/players/" + encodeURIComponent(this.props.playerName);

		let result = (

				<Link to={playerNameParam} className="player-results-item-wrapper react-link" onClick={() => this._handleClick(this.props.playerName)} >

					<span className={"club-shirt player-result-player-shirt " + club_shirts[this.props.playerClub]}> </span>
					<p className="player-result-player-name heading-text black-text"> {this.props.playerName} </p>
					<p className="player-result-player-position body-text-light black-text"> {this.props.playerPosition} </p>
					<p className="player-result-player-club body-text-light black-text"> {this.props.playerClub} </p>
					<ReviewScoreCircle reviewScore={this.props.reviewScore}/>
					<p className="player-result-view-player body-text-light black-text">VIEW PLAYER</p>
					<i className="arrow-player-result-item fa fa-long-arrow-right" aria-hidden="true"></i>

				</Link>
			);

		return result;
	}
}


// ===============
// Connect redux state container to PlayerResult component
// ===============


const mapStateToProps = state => {
  return { 

	  	playersViewed: state.playersViewed
  	};
};



// Export connected component
export const PlayerResultContainer = connect(mapStateToProps, { setPlayer })(PlayerResult);



