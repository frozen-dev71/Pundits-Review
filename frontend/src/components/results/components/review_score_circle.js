// ======================================
// Review Score Circle Component - Contains colored circle with review score
// ======================================


// ===========
// IMPORTS 
// ===========

import React, { Component } from "react";

// Local JS Imports 

// CSS Imports
import './review_score_circle.css';




export class ReviewScoreCircle extends Component {

	constructor(props) {
    	super(props);
    	this.state = {
      		backgroundClass: " background-light",
   		};

   		this._handleBackgroundSet = this._handleBackgroundSet.bind(this);
  	}

  	componentDidMount() {

  		this._handleBackgroundSet();
  	}

    componentDidUpdate() {

      if (this.state.backgroundClass !== this._setBackground()) {
        this._handleBackgroundSet();
      }
    }

    _handleBackgroundSet() {

      this.setState(prevState => ({
          backgroundClass: this._setBackground()
      }));

    }

  	_setBackground() {

      if (this.props.reviewScore !== "N/A") {

          if (this.props.reviewScore >= 70 && this.props.reviewScore < 101) {
          return ' green-circle';
        } 

          if (this.props.reviewScore >= 40  && this.props.reviewScore < 70) {
            return ' yellow-circle';
          } 

          if (this.props.reviewScore >= 0  && this.props.reviewScore < 40) {
            return ' red-circle';
          }

      } else {

        return ' black-circle';

      }

  	}


	render() {

		let result = <div className={"review-score-circle body-text-light white-text" + this.state.backgroundClass}>{this.props.reviewScore}</div>

		return result;
	}
}

