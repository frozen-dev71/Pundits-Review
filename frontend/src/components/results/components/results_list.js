// ======================================
// Results List Component to access data from Redux State
// Renders a result item for every player or club returned from the API - depending on the view selected
// ======================================


// ===========
// IMPORTS 
// ===========

import React, { Component } from "react";
import { connect } from "react-redux";

// Local JS Imports
import { searchByClub, searchByPlayer } from "../../actions/actions";
import { PlayerResultContainer } from './player_result_item';
import { ClubResultContainer } from './club_result_item';
import { DividingLine } from  "../../global/components/dividing_line";
import { setReviewScore } from "../../global/functions/set_data_helpers";
import { NoResults } from '../../global/components/no_results_found';
import { ReviewScoreExplainer } from './review_score_explainer';

// CSS Imports
import './result_list.css';

// ===========
// Results Component 
// ===========

export class Results extends Component {

	constructor(props) {
    	super(props);
    	this.state = {
      		playerViewSelected: true,
   		};

   		this._handleUpdate = this._handleUpdate.bind(this);
  	}

  	componentDidUpdate() {
    	
    	this._handleUpdate();

  	}

  	_handleUpdate() {

  		// Toggle the playerViewSelected state
  		if (this.props.playerViewSelected !== this.state.playerViewSelected) {

  			this.setState(prevState => ({
				playerViewSelected: this.props.playerViewSelected
			}));

  		}
  	}

    _setReviewScore(el, clubOrPlayer) {

      return setReviewScore(el, clubOrPlayer)[0];
    }

  	render() {

  		let result;

  			// PLAYER VIEW 
  			if (this.props.playerViewSelected) {

          if (this.props.searchDataPlayers.length) {

              result = (

              <div className="results-list-wrapper">

                <ReviewScoreExplainer />

                {this.props.searchDataPlayers.map((el, index) => {


                    let playerItem = (

                        <div key={index} className="item-and-line-wrapper">

                          <PlayerResultContainer playerName={el.player} playerPosition={el.position} playerClub={el.club_name} reviewScore={this._setReviewScore(el, false)}/>
                          <DividingLine/>

                        </div>

                      );

                    return playerItem;

                })}

            </div>

            );

          } else {

              result = (

                  <div className="results-list-wrapper">

                    <NoResults playerOrClub={"players"} searchQuery={this.props.searchTerms[this.props.searchTerms.length -1]}/>

                  </div>  

                )

          }

			// CLUB VIEW 
  			} else {

          if (this.props.searchDataClubs.length) {

              result = (

                  <div className="results-list-wrapper">

                    <ReviewScoreExplainer />

                    {this.props.searchDataClubs.map((el, index) => {

                      let clubItem = (

                            <div key={index} className="item-and-line-wrapper">

                              <ClubResultContainer clubName={el.club} reviewScore={this._setReviewScore(el, true)}/>
                              <DividingLine/>
                              
                            </div>

                        );
                        
                        return clubItem;

                    })}

                </div>

                );

          } else {

              result = (

                  <div className="results-list-wrapper">

                    <NoResults playerOrClub={"clubs"} searchQuery={this.props.searchTerms[this.props.searchTerms.length -1]}/>

                  </div>  

                )

          }


  			}

  		return result;
  	}


}

// ===============
// Connect redux state container to Results component
// ===============

const mapStateToProps = state => {
  return { 
  		searchDataClubs: state.searchDataClubs,
  		searchDataPlayers: state.searchDataPlayers,
      searchTerms: state.searchTerms
  	};
};



export const ResultsList = connect(mapStateToProps)(Results);
