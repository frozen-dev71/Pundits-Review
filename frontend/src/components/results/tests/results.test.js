// ======================================
// This file tests for the components listed in the RESULTSPAGE sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that ResultsPage renders correctly
// 2. Check that ClubResultContainer renders correctly
// 3. Check that PlayerResultContainer renders correctly
// 4. Check that PlayerClubToggle renders correctly
// 5. Check that ResultsList renders correctly
// 6. Check that ReviewScoreCircle renders correctly
// 7. Check that ReviewScoreExplainer renders correctly
// 8. Check that ThisWeekEveryoneToggle renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { ResultsPage } from '.././resultspage';
import { ClubResultContainer } from '.././components/club_result_item';
import { PlayerResultContainer } from '.././components/player_result_item';
import { PlayerClubToggle } from '.././components/playerclubtoggle';
import { ResultsList } from '.././components/results_list';
import { ReviewScoreCircle } from '.././components/review_score_circle';
import { ReviewScoreExplainer } from '.././components/review_score_explainer';
import { ThisWeekEveryoneToggle } from '.././components/thisweek_everyone_toggle';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Results Page Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that ResultsPage renders correctly
	// ===========

	describe('ResultsPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ResultsPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that ClubResultContainer renders correctly
	// ===========

	describe('ClubResultContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ClubResultContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that PlayerResultContainer renders correctly
	// ===========

	describe('PlayerResultContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PlayerResultContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that PlayerClubToggle renders correctly
	// ===========

	describe('PlayerClubToggle', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PlayerClubToggle/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that ResultsList renders correctly
	// ===========

	describe('ResultsList', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ResultsList/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 6. Check that ReviewScoreCircle renders correctly
	// ===========

	describe('ReviewScoreCircle', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ReviewScoreCircle/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 7. Check that ReviewScoreExplainer renders correctly
	// ===========

	describe('ReviewScoreExplainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ReviewScoreExplainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 8. Check that ThisWeekEveryoneToggle renders correctly
	// ===========

	describe('ThisWeekEveryoneToggle', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ThisWeekEveryoneToggle/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



});
