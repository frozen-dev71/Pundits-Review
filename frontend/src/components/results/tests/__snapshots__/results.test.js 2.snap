// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Page Component Tests ClubResultContainer should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ClubResult) />
  </Provider>
</Router>
`;

exports[`Results Page Component Tests PlayerClubToggle should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(PlayerClubToggleComponent) />
  </Provider>
</Router>
`;

exports[`Results Page Component Tests PlayerResultContainer should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(PlayerResult) />
  </Provider>
</Router>
`;

exports[`Results Page Component Tests ResultsList should render correctly 1`] = `
<div
  class="results-list-wrapper"
>
  <div
    class="no-results-wrapper"
  >
    <p
      class="no-results-text body-text-light"
    >
      We couldn't find any clubs matching " "
    </p>
    <div
      class="dividing-line background-dark"
    />
    <p
      class="no-results-sub-text body-text-light"
    >
      It doesn't look like we have data this week for ' '. Try looking under 'ALL WEEKS' to see if we have older results matching your search.
    </p>
  </div>
</div>
`;

exports[`Results Page Component Tests ResultsPage should render correctly 1`] = `
<div
  class="results-page-wrapper"
>
  <div
    class="search-bar-results-wrapper side-margin"
  >
    <div
      class="search-bar-wrapper background-light"
    >
      <input
        aria-label="Search for a player or a club"
        aria-required="true"
        autocomplete="off"
        class="search-box body-text-light"
        id="search-bar-input"
        name="Search input to search for Premier League players or clubs."
        placeholder="Search for a player or a club ..."
        type="text"
        value=""
      />
      <a
        class="react-link"
        href="/results"
      >
        <button
          class="search-button"
          id="search-button"
          type="submit"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </a>
      <span />
    </div>
    <p
      class="results-info-text body-text-light"
    >
      0 results found for ""
    </p>
  </div>
  <div
    class="results-menu-wrapper"
  >
    <div
      class="player-club-toggle-wrapper"
    >
      <div
        class="player-club-toggle-player-inner-container toggle-inner-container toggle-active"
        id="player-toggle"
        role="button"
      >
        <p
          class="player-club-menu-text heading-text"
        >
          Players
        </p>
        <div
          class="results-number-box number-box-active"
          id="player-number-box"
        >
          <p
            class="results-number-text body-text"
          >
            0
          </p>
        </div>
      </div>
      <div
        class="player-club-toggle-club-inner-container toggle-inner-container"
        id="club-toggle"
        role="button"
      >
        <p
          class="player-club-menu-text heading-text"
        >
          Clubs
        </p>
        <div
          class="results-number-box"
          id="club-number-box"
        >
          <p
            class="results-number-text body-text"
          >
            0
          </p>
        </div>
      </div>
    </div>
    <div
      class="this-week-everyone-toggle-wrapper"
    >
      <div
        class="this-week-everyone-button-container"
      >
        <button
          class="this-week-everyone-button this-week-everyone-button-active body-text-light"
          id="this-week-button"
        >
          THIS WEEK
        </button>
        <button
          class="this-week-everyone-button body-text-light"
          id="everyone-button"
        >
          ALL WEEKS
        </button>
      </div>
    </div>
    <div
      class="results-list-wrapper"
    >
      <div
        class="no-results-wrapper"
      >
        <p
          class="no-results-text body-text-light"
        >
          We couldn't find any players matching " "
        </p>
        <div
          class="dividing-line background-dark"
        />
        <p
          class="no-results-sub-text body-text-light"
        >
          It doesn't look like we have data this week for ' '. Try looking under 'ALL WEEKS' to see if we have older results matching your search.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Results Page Component Tests ReviewScoreCircle should render correctly 1`] = `
<div
  class="review-score-circle body-text-light white-text background-light"
/>
`;

exports[`Results Page Component Tests ReviewScoreExplainer should render correctly 1`] = `
<div
  class="review-score-explainer-wrapper"
>
  <div
    class="review-score-circle body-text-light white-text background-light"
  >
    100
  </div>
  <p
    class="review-score-explainer-text body-text-light"
  >
    THIS WEEKS PR SCORE 
  </p>
</div>
`;

exports[`Results Page Component Tests ThisWeekEveryoneToggle should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ThisWeekEveryoneToggleContainer) />
  </Provider>
</Router>
`;
